### PR TITLE
Feature/implement type script

### LINKED ISSUE #1: Independent Research Spike- Friday
share findings Sat

### LINKED ISSUE #2: Day 3 Deliverables
- [ ] In a DM with both instructors, send us the link to a recent PR

- [ ] Choose a PR that is related to your Stretch Tech, and use the PR description to fill us in on your learning.

- [ ] Sometime in the next few days, we’ll perform a code review

- [ ] You do not need to wait for our review to merge it in.

### PR DESCRIPTION
**Pull request by kendallm360**
## Summary:
Implements initial TypeScript understanding in the App, RandomCard and RandomContainer Components. Assigned "Types/Interfaces" to all simple datatypes and objects.

## Type of Changes:
- [x] Bug fix
- [x] Refactor
- [x] New feature

## Where are the Changes:
All files (initial PR)

## How was this tested:
Chrome Dev tools
React Component Tree
Console/Virtual Dom

## What are the relevant tickets:
closes #2 
closes #1

### CODE REVIEW

**Review by nhodnett — APPROVED** [review]
Nice work on this initial setup crew! This was a little harder than I thought it would be, but I can see how the learning curve might be steep and then ease of a little... We got this!

File: src/components/RandomContainer/RandomContainer.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ type Drink = {
 
 const RandomContainer: FC<RandomDrinks> = ({ randomDrinks }) => {
```
Comment (jordan-sullivan) [inline src/components/RandomContainer/RandomContainer.tsx:15]: I think there might be a way we can refactor line 15  to not even need the `randomDrinks` prop as we have the `Drinks ` Type now. we may be able to bypass this step.

File: src/components/randomDrinkCards/RandomDrinkCards.tsx [hunk 2]
```diff
@@ -7,11 +7,14 @@ type Drink = {
...
-const RandomDrinkCards: FC<Drink> = () => {
+const RandomDrinkCards: FC<Drink> = ({ strDrink, strDrinkThumb, idDrink }) => {
   return (
     <div className="drink-card">
-      <h2>Stuff will be here</h2>
-      {/* <img/> */}
+      <h2>{strDrink}</h2>
+      <img
+        src={strDrinkThumb}
+        alt={`Picture${idDrink} of a cocktail stirred to perfection`}
```
Comment (jordan-sullivan) [inline src/components/randomDrinkCards/RandomDrinkCards.tsx:16]: Really great thought to include an `alt tag` for images to improve accessibility! Wondering if we should use the title of the drink interpolated here ( `strDrinkThumb` ) as well to be even more detailed on the picture description?

**Review by jordan-sullivan — COMMENTED** [review]
Nice work on the initial implementation of TS team! Much more to research but off to a great start. We will need to be sure to create randomize function and only display 4 cards in next branch, but glad to see everything is rendering successfully! - Jordan

File: src/apiCalls.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  return fetch("https://www.thecocktaildb.com/api/json/v1/1/random.php").then(
+    (response) => response.json()
+  );
+};
+
+const fetchAllCocktails = () => {
+  return fetch(
+    "https://www.thecocktaildb.com/api/json/v1/1/filter.php?g=Cocktail_glass"
+  ).then((response) => response.json());
+};
```
Comment (sertmer) [inline src/apiCalls.tsx:11]: Aside from the URL here, these fetch calls look really similar. Can you think of any way to DRY these up?

File: src/components/App/App.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    // });
+    fetchAllCocktails().then((data) => {
+      this.setState({
+        drinkArray: data.drinks,
+      });
+    });
+  };
+
+  render() {
+    // console.log(this.state.drinkArray);
```
Comment (sertmer) [inline src/components/App/App.tsx:46]: Lets be sure to clear comments before we push up!

**Review by sertmer — COMMENTED** [review]
I left a few comments for y'all, but all in all, it looks like you're in a good spot! Great work setting this up!

**Pull request merged**